tests/functional/core/worker/warm_start/test_warm_starting.py

```diff
@@ -0,0 +1,225 @@
+from orion.storage.base import BaseStorageProtocol, setup_storage
+
+# Function to create a space.
+_space: Callable[[dict], Space] = SpaceBuilder().build
+
+
+def simple_quadratic(
+    a: float = 1, b: float = -2, c: float = 1
+) -> Callable[[float], float]:
+    """A simple quadratic function."""
+    return lambda x: a * x**2 + b * x + c
+
+
+def _temp_storage(pickle_path: Path | str) -> BaseStorageProtocol:
+    """Creates a temporary storage."""
+    return setup_storage(
+        {
+            "type": "legacy",
+            "database": {"type": "pickleddb", "host": str(pickle_path)},
+        }
+    )
+
+
+@pytest.mark.parametrize("algo", [TPE])
+def test_warm_starting_helps(
+    algo: type[BaseAlgorithm],
+    storage: BaseStorageProtocol,
+):
+    """Integration test. Shows that warm-starting helps in a simple task."""
+    algo_config = {"of_type": algo.__qualname__.lower(), "seed": 42}
+
+    # TODO: There still seems to be some randomness in this test, despite the algorithms being
+    # seeded..
+    np.random.seed(123)
+    random.seed(123)
+
+    source_task = simple_quadratic(a=1, b=-2, c=1)
+    target_task = simple_quadratic(a=1, b=-2, c=1)
+    # Note: minimum is at x = 1, with y = 0
+    # Here we could prime the source task with points that should be super useful, since they
+    # are really close to the minimum:
+    source_space: Space = _space({"x": "uniform(0, 10)"})
+    target_space: Space = _space({"x": "uniform(0, 10)"})
+
+    n_source_trials = 50  # Number of trials from the source task
+    max_trials = 10  # Number of trials in the target task
+
+    source_experiment = build_experiment(
+        name="source_exp",
+        space=source_space,
+        algorithms={"of_type": "random", "seed": 42},
+        storage=storage,
+        max_trials=n_source_trials,
+    )
+    source_experiment.workon(_wrap(source_task))
+    # Create the Knowledge base by passing the now-filled Storage.
+    knowledge_base = KnowledgeBase(storage=storage)
+    assert knowledge_base.n_stored_experiments == 1
+
+    without_warm_starting = workon(
+        _wrap(target_task),
+        name="witout_warm_start",
+        space=target_space,
+        max_trials=max_trials,
+        algorithms=algo_config,
+        knowledge_base=None,
+    )
+    assert len(without_warm_starting.fetch_trials()) == max_trials
+
+    with_warm_starting = workon(
+        _wrap(target_task),
+        name="with_warm_start",
+        space=target_space,
+        max_trials=max_trials,
+        algorithms=algo_config,
+        knowledge_base=knowledge_base,
+    )
+    assert len(with_warm_starting.fetch_trials()) == max_trials
+
+    best_trial_with = _get_best_trial(with_warm_starting)
+    best_trial_without = _get_best_trial(without_warm_starting)
+
+    objective_with = _get_objective(best_trial_with)
+    objective_without = _get_objective(best_trial_without)
+    # FIXME: This is still somehow not always true, weirdly enough. Perhaps the TPE algo has
```

Comment:
Did you manage to solve this issue?

Reply:
Didn't change anything really, but strangely, I've been running the tests like 10 times this morning and it's passed every time.  :thinking: